tests/unit/test_key.py

```diff
@@ -212,6 +212,24 @@ def test_constructor_with_namespace():
         )
         assert key._reference is None
 
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test_constructor_with_default_namespace_as_empty_string(context):
+        context.client.namespace = "DiffNamespace"
+        key = key_module.Key("Kind", 1337, namespace="")
+
+        assert key._key == google.cloud.datastore.Key("Kind", 1337, project="testing")
+        assert key._reference is None
```

Comment:
Shouldn't we be asserting that the key's namespace matches `context.client.namespace` here (and below)?

Reply:
So, the term "default" is somewhat overloaded here, which is leading to some confusion. Here, it specifically means the no-name namespace that is the default for Datastore, not some other user-set "default". In passing the empty string here (and `None` below), we're explicitly setting that no-name namespace independently of what may have been set on the client or context.

Reply:
OK, so does the equality test (in line 221) actually verify that `key._key.namespace is None` as expected?

Reply:
Ah, right. Nope. 

d9f11fa

I also fixed an immediate neighbor that was also not testing what it said it was.